projects/igniteui-angular/src/lib/directives/date-time-editor/date-time-editor.directive.ts

```diff
@@ -301,30 +327,28 @@ export class IgxDateTimeEditorDirective extends IgxMaskDirective implements OnCh
    *
    * @param datePart The optional DatePart to increment. Defaults to Date or Hours(when Date is absent from the inputFormat - ex:'HH:mm').
    */
-  public increment(datePart?: DatePart): void {
+  public increment(datePart?: DatePart, spinDeltas?: DatePartDeltas): void {
+    const deltas = Object.assign({}, this.datePartDeltas, spinDeltas);
     const targetDatePart = this.targetDatePart;
     if (!targetDatePart) {
```

Comment:
Oh boy, looking at what's going on in `targetDatePart`, this should not be the check here. The target date part should be evaluated between the optional parameter first, the current target (and a default)